crates/telemetry/src/lib.rs

```diff
@@ -16,9 +17,35 @@ pub use propagation::inject_trace_context;
 /// Initializes telemetry for Spin using the [tracing] library.
 ///
 /// Under the hood this involves initializing a [tracing::Subscriber] with multiple [Layer]s. One
-/// [Layer] emits [tracing] events to stderr, and another sends spans to an OTEL collector.
+/// [Layer] emits [tracing] events to stderr, another sends spans to an OTel collector, and another
+/// sends metrics to an OTel collector.
 ///
-/// Configuration is pulled from the environment.
+/// Configuration for the OTel layers is pulled from the environment.
+///
+/// Examples of emitting traces from Spin:
+///
+/// ```no_run
+/// #[instrument(name = "span_name", err(level = Level::INFO), fields(otel.name = "dynamically set name"))]
+/// fn func_you_want_to_trace() {}
+/// ```
+///
+/// Some notes on tracing:
+///
+/// - If you don't want the span to be collected by default emit it at a trace or debug level.
+/// - Make sure you `.in_current_span()` any spawned tasks so the span context is propagated.
+/// - Use the otel.name attribute to dynamically set the span name.
+/// - Use the err argument to have instrument automatically handle errors.
+///
+/// Examples of emitting metrics from Spin:
+///
+/// ```no_run
+/// tracing::trace!(monotonic_counter.metric_foo = 1, attribute = "value");
```

Comment:
It might be useful to give more concrete examples and/or explain them.  It seems like the number is the "amount" of metric to be recorded (e.g. request count, egress bytes, etc.), is that right?  And `attribute` is actually the name under which `value` is to be associated with this amount?  E.g. it could be a `url = ...` to count requests to a given URL, or `app_name = ...` to count requests to a given app?

Reply:
> it could be a url = ... to count requests to a given URL, or app_name = ... to count requests to a given app?

Yep; "attribute" is opentelemetry parlance, sometimes called "tag" in other systems.